### DISCUSSION TITLE
建议：GUI界面增加对鼠标的支持

### DISCUSSION THREAD

**Discussion by xiwangly2** [ventoy/Ventoy#1433] [Ideas]
有时候，我并不希望总是使用键盘来选择菜单的选项以及使用功能键使用那些功能，我希望能够使用鼠标来选择它们。可以通过插件提供支持。

**Comment by ventoy** [comment]
难度以及工作量都超出了我的能力范围。

**Comment by iPixelOldC** [comment]
这得grub2支持鼠标操作吧，但貌似grub2至今都没支持，但好像有第三方的[支持](https://bitbucket.org/byteit101/grub-mouse/)。不过看上去12年没更新了，应该已经弃坑了。老哥，放弃吧。这事grub2的支持不解决，这里也没办法解决2333 不过可以去瞅瞅grub2有没有相关的pr，然后追随下更新情况2333

**Comment by a1ive** [comment]
https://github.com/ventoy/Ventoy/pull/1457
UEFI下简单的鼠标支持 (鼠标映射键盘)
测试方法：GRUB2 界面按`c`进入控制台执行 `terminal_input --append mouse`
按键映射：
鼠标左键 -> Enter
鼠标右键 -> Esc
鼠标上移 -> 方向键上
鼠标下移 -> 方向键下
可以在这里下载测试：
https://github.com/a1ive/Ventoy/actions/runs/1862476726

**Reply by StarRing3906** [reply]
打不开啊

**Reply by a1ive** [reply]
> 打不开啊

pr已经被合并了。你可以直接在 Ventoy 里面测试。

**Reply by StarRing3906** [reply]
？我试了，鼠标没用啊

**Reply by a1ive** [reply]
> ？我试了，鼠标没用啊

是否按pr里面的说明进行操作？你电脑的UEFI是否支持鼠标？

**Reply by StarRing3906** [reply]
> > ？我试了，鼠标没用啊
> 
> 是否按pr里面的说明进行操作？你电脑的UEFI是否支持鼠标？

哦，对不起，没看到要按C键执行 terminal_input--append mouse 
等下，我触屏怎么按C键，还是不用Ventoy了罢（悲

**Reply by a1ive** [reply]
> > > ？我试了，鼠标没用啊
> > 
> > 
> > 是否按pr里面的说明进行操作？你电脑的UEFI是否支持鼠标？
> 
> 哦，对不起，没看到要按C键执行 terminal_input--append mouse 等下，我触屏怎么按C键，还是不用Ventoy了罢（悲

编辑 EFI 分区里面的 grub.cfg，这里去掉注释就行。
https://github.com/ventoy/Ventoy/blob/82053680bf438f9da570d393315fcbbd9b752e25/INSTALL/grub/grub.cfg#L2580
另外，只支持鼠标和触摸板 (Touchpad/Trackpad)， 不支持触摸屏 (Touchscreen) ，它们不是同一类设备。
